central/reportconfigurations/service/v2/service_impl.go

```diff
@@ -46,7 +53,7 @@ func (*serviceImpl) AuthFuncOverride(ctx context.Context, fullMethodName string)
 func (s *serviceImpl) PostReportConfiguration(ctx context.Context, request *apiV2.ReportConfiguration) (*apiV2.ReportConfiguration, error) {
 	protoReportConfig := reportConfigConverter.ConvertV2ReportConfigurationToProto(request)
 	if err := s.validator.ValidateReportConfiguration(ctx, protoReportConfig); err != nil {
-		return nil, err
+		return nil, errors.Errorf("Report config validation failed : %s", err)
```

Comment:
`errors.Wrap()`

Reply:
I avoided `errors.Wrap()` here because that requires an error type such as `invalidArgs` and the validator can also throw error for reasons like error from storage layer. So it won't strictly be `invalidArgs` each time.

Reply:
The first argument to `errors.Wrap()` is expected to be of error type. `invalidArgs` is simply a sentinal error. Your error formatting is similar to `errors.Wrap` cascading that comes OOTB.
Recommend changing it to `errors.Wrap(err, "validating report configuration")`

Reply:
Done